app/controllers/questions_controller.rb

```diff
@@ -19,9 +23,111 @@ def show
       gon.solution = solution
     end
 
+    if !(current_user.show_solutions)
+      if @room.answered(@question.id,@room.id)
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id) + 1
+      else
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id)
+      end
+    else
+      @number_of_question = @room.learning_objects.where("learning_object_id < ?",@question.id).count + 1
```

Comment:
Ano usporiadala som ich podla idcka, ale len v rezime, ked sa zobrazuju spravne odpovede. V tomto rezime som nechcela menit poradie otazok, prave z toho dovodu, aby sa lepsie ucili. Skusila som ich zobrazovat nahodne, ale nebolo to dobre. Takze chcem zachovat jedno poradie, v ktorom sa v tomto rezime budu zobrazovat. Mozem ich pripadne usporiadat na zaklade narocnosti, ale neviem ci je to nutne. Vzhladom na ucel tohto rezimu si myslim, ze je jedno v akom poradi su, dolezite je aby sa to poradie nemenilo. Co si o tom myslite? Mam to nejako prerobit?